packages/protocol/contracts/thirdparty/AddressManager.sol

```diff
@@ -85,23 +87,14 @@ contract AddressManager is OwnableUpgradeable {
 
     /**
      * Retrieves the address associated with a given name.
-     * @param _name Name to retrieve an address for.
+     * @param name Name to retrieve an address for.
      * @return Address associated with the given name.
      */
-    function getAddress(string memory _name) external view returns (address) {
-        return addresses[_getNameHash(_name)];
-    }
-
-    /**********************
-     * Internal Functions *
-     **********************/
-
-    /**
-     * Computes the hash of a name.
-     * @param _name Name to compute a hash for.
-     * @return Hash of the given name.
-     */
-    function _getNameHash(string memory _name) internal pure returns (bytes32) {
-        return keccak256(abi.encodePacked(_name));
+    function getAddress(string memory name) external view returns (address) {
+        bytes32 nameHash;
+        assembly {
+            nameHash := keccak256(add(name, 32), mload(name))
```

Comment:
Lets compare the gas usage of `nameHash = keccak256(bytes(name))` please.

Reply:
According this comparision: https://github.com/taikoxyz/taiko-mono/blob/c4bc252a2dbd2e7f4a25009e8ff7e7e2b096f646/packages/protocol/test2/GasComparison.t.sol#L26
`keccak256(bytes(name))` is cheaper.

| test2/GasComparison.t.sol:FooBar contract |                 |     |        |     |         |
|-------------------------------------------|-----------------|-----|--------|-----|---------|
| Deployment Cost                           | Deployment Size |     |        |     |         |
| 71123                                     | 387             |     |        |     |         |
| Function Name                             | min             | avg | median | max | # calls |
| hashString_1                              | 631             | 631 | 631    | 631 | 1       |
| hashString_2                              | 609             | 609 | 609    | 609 | 1       |